src/main/java/amazon/AllSearchResultPage.java

```diff
@@ -33,10 +35,16 @@ public boolean textVisibilityWhenClickOnBestSellerButton() {
         Random random=new Random();
         int randomNumber=random.nextInt(1, listItemWhereWriteBestSeller.size());
         Actions act=new Actions(driver);
-        act.moveToElement(listItemWhereWriteBestSeller.get(randomNumber)).build().perform();
-        boolean textVisibility = textListWhenClickOnBestSellerButton.get(randomNumber).getText().contains("in");
+        boolean textVisibility=false;
+       try {
+           act.moveToElement(listItemWhereWriteBestSeller.get(randomNumber)).build().perform();
+           textVisibility = textListWhenClickOnBestSellerButton.get(randomNumber).getText().contains("in");
+       }catch (NoSuchElementException e){
 
-        return textVisibility;
+           driver.close();
```

Comment:
Why you close the driver here? What's the meaning behind it? Closing driver doesn't end the test. If you do nothing in case of best seller items absence, then the test will end at that point. Closing driver in Page Object is a bad practice for many reasons, you don't need to do that.